velox/exec/fuzzer/JoinFuzzer.cpp

```diff
@@ -597,6 +603,12 @@ std::optional<core::JoinType> tryFlipJoinType(core::JoinType joinType) {
 // Returns a plan with flipped join sides of the input hash join node. If the
 // join type doesn't allow flipping, returns a nullptr.
 core::PlanNodePtr tryFlipJoinSides(const core::HashJoinNode& joinNode) {
+  //  Null-aware right semi project join doesn't support filter.
+  if (joinNode.filter() &&
+      joinNode.joinType() == core::JoinType::kLeftSemiProject &&
+      joinNode.isNullAware()) {
+    return nullptr;
+  }
```

Comment:
Could you share an example query of this kind and why it doesn't support filter? By the way, I noticed that the code checks joinNode.joinType() == core::JoinType::kLeftSemiProject but the comment says "right semi project join".

Reply:
I don't know why it isn't supported. I came across the edge case here: 
https://github.com/facebookincubator/velox/blob/db0c3e697cad660dffa96e6af3880395f6540f90/velox/core/PlanNode.h#L1634

This method flips the join type, so my change prevents left semi project from being flipped to right and instead return nullptr.